### PR TITLE
Alerting: Fix incorrect display of pending period in alert rule form

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**

Fix for the display of pending period on alert rules. The correct value was saved/retrieved from the backend, but just shown incorrectly in the UI when editing rules.

In addition, also fixes the edit group modal not updating the dirty state of the form when clicking one of the quick eval buttons, and also disables the save button if the form is invalid

Introduced by https://github.com/grafana/grafana/pull/85010

**Steps to reproduce**

- Edit an existing Grafana Managed alert rule
- Change the pending period from 1m to 5m
- Save changes
- Edit the alert rule again
- Observe that it still says 1m.
- Expanding the alert instance row shows the 5m pending period.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx [hunk 1]
```diff
@@ -234,7 +234,7 @@ export function EditCloudGroupModal(props: ModalProps): React.ReactElement {
     handleSubmit,
     register,
     watch,
-    formState: { isDirty, errors },
+    formState: { isDirty, errors, isValid },
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx:237]: Strangely, just destructuring the `isValid` is enough to trigger `act` warnings in the tests for this component - even if it isn't used 🤷 
I haven't dived into why as I don't have the chance to right now, but I updated the tests regardless to wait for the component to settle before asserting

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM! :shipit: 
tested locally and it works perfect!

File: public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx (hunk 1 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx:237]: 🤔

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx:237]: something to investigate when we have time 😊

**Pull request merged**